src/main.rs

```diff
@@ -77,6 +78,21 @@ fn main() -> glib::ExitCode {
                 application_opts.test_dc = true;
             }
 
+            let maybe_client_id = dict.lookup::<String>("client-id").unwrap();
+            let maybe_client_secret = dict.lookup::<String>("client-secret").unwrap();
+
+            match (maybe_client_id, maybe_client_secret) {
+                (Some(client_id), Some(client_secret)) => {
+                    application_opts.client_id = client_id.parse::<i32>().unwrap();
+                    application_opts.client_secret = Cow::Owned(client_secret);
+                }
+                (None, None) => (),
+                _ => {
+                    log::error!("Both client-id and client-secret must be set together");
+                    return -1;
```

Comment:
You should return a positive number here in order to exit the application.
See https://docs.gtk.org/gio/signal.Application.handle-local-options.html